### PR TITLE
Modified scripts to optionally enable helm tls

### PR DESCRIPTION
**Pull request by adusumillipraveen**
### JIRA link (if applicable) ###

https://dev.azure.com/hmcts/Platform%20Engineering/_workitems/edit/39

### Change description ###

Extracted scripts from https://dev.azure.com/hmcts/_git/CNP?path=%2Fhelm-tls.yaml&version=GBmaster
Reordered arguments , knowing that it will break. Will fix it in UI before merging this PR.  

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: scripts/install-flux.sh [hunk 1]
```diff
@@ -3,10 +3,21 @@ set -ex
 
 ENV=$1
 CLUSTER_NAME=$2
-GIT_REPO=${3:-https://weaveworks.github.io/flux}
-VALUES=${4-deployments/weave-flux/values.yaml}
+VAULT_NAME=$3
+ENABLE_HELM_TLS=${4:false}
```
Comment (timwebster9) [inline scripts/install-flux.sh]: why do we need this?

File: scripts/generate-helm-pki.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+echo subjectAltName=IP:127.0.0.1 > extfile.cnf
+
+openssl x509 -req -CA ca.cert.pem -CAkey ca.key.pem -CAcreateserial \
+-in tiller.csr.pem -out tiller.cert.pem -days 365 -extfile extfile.cnf
+
+openssl x509 -req -CA ca.cert.pem -CAkey ca.key.pem -CAcreateserial \
+-in helm.csr.pem -out helm.cert.pem -days 365
+
+# Generate secret container Helm certs for Flux while we're at it
```
Comment (timwebster9) [inline scripts/generate-helm-pki.sh]:
```suggestion
# Generate secret for Helm certs for Flux while we're at it
```

File: scripts/install-flux.sh (hunk 1 above)
Comment (adusumillipraveen) [reply scripts/install-flux.sh]: To override in pipeline to enable for only environments where we want this

Comment (timja) [reply scripts/install-flux.sh]: sandbox and preview aren't supposed to have it enabled

I wouldn't default it though, I think it should always be set.
I don't like these optional arguments, we've already been bitten once by it

Comment (timwebster9) [reply scripts/install-flux.sh]: 👍

Comment (adusumillipraveen) [reply scripts/install-flux.sh]: Sure, i will get rid of the default in both scripts.

File: scripts/install-flux.sh [hunk 3]
```diff
@@ -3,10 +3,21 @@ set -ex
 
 ENV=$1
 CLUSTER_NAME=$2
-GIT_REPO=${3:-https://weaveworks.github.io/flux}
-VALUES=${4-deployments/weave-flux/values.yaml}
+VAULT_NAME=$3
+ENABLE_HELM_TLS=$4
+GIT_REPO=${5:-https://weaveworks.github.io/flux}
```
Comment (timja) [inline scripts/install-flux.sh]: i don't think you should be able to override these, it's just going to cause issues when someone adds a new param, i.e. you updating this is going to break it

File: scripts/install-flux.sh [hunk 4]
```diff
@@ -3,10 +3,21 @@ set -ex
...
+GIT_REPO=https://weaveworks.github.io/flux
+VALUES=deployments/weave-flux/values.yaml
 
 helm repo add weaveworks ${GIT_REPO}
+
+if [[ ${ENABLE_HELM_TLS} == true ]]
+then
+ az keyvault secret download --vault-name ${VAULT_NAME} --encoding ascii --name helm-pki-flux-secret --file helm-client-certs.yaml
+ kubectl apply -f helm-client-certs.yaml
+ params+=(--tls --tls-verify --tls-ca-cert ca.cert.pem --tls-cert helm.cert.pem --tls-key helm.key.pem  --set helmOperator.tls.caContent="$(cat ca.cert.pem)" )
```
Comment (timja) [inline scripts/install-flux.sh]:
```suggestion
 helm_tls_params+=(--tls --tls-verify --tls-ca-cert ca.cert.pem --tls-cert helm.cert.pem --tls-key helm.key.pem  --set helmOperator.tls.caContent="$(cat ca.cert.pem)" )
```

File: scripts/install-flux.sh [hunk 5]
```diff
@@ -3,10 +3,21 @@ set -ex
...
+fi
+
 helm upgrade flux weaveworks/flux --install --namespace admin -f ${VALUES} \
     --set "git.path=k8s/${ENV}/common\,k8s/${ENV}/${CLUSTER_NAME}\,k8s/common",git.label=${ENV},git.email=flux-${ENV}@hmcts.net,git.user="Flux ${ENV}" \
-    --wait
+    "${params[@]}"
```
Comment (timja) [inline scripts/install-flux.sh]:
```suggestion
    "${helm_tls_params[@]}"
```

File: scripts/install-tiller.sh [hunk 6]
```diff
@@ -3,7 +3,14 @@
...
+ENABLE_HELM_TLS=$4
+RBAC_CONFIG=roles/helm-rbac-config.yaml
 
 kubectl create -f ${RBAC_CONFIG}
-helm init --wait --service-account tiller --tiller-image gcr.io/kubernetes-helm/tiller:v${HELM_VERSION} --history-max=${HISTORY_MAX} --kube-context ${CLUSTER_NAME}-admin
\ No newline at end of file
+
+if [[ ${ENABLE_HELM_TLS} == true ]]
+then
+ params+=(--tiller-tls --tiller-tls-cert ./tiller.cert.pem --tiller-tls-key ./tiller.key.pem --tiller-tls-verify --tls-ca-cert ./ca.cert.pem)
```
Comment (timja) [inline scripts/install-tiller.sh]:
```suggestion
 helm_tls_params+=(--tiller-tls --tiller-tls-cert ./tiller.cert.pem --tiller-tls-key ./tiller.key.pem --tiller-tls-verify --tls-ca-cert ./ca.cert.pem)
```

File: scripts/install-tiller.sh [hunk 7]
```diff
@@ -3,7 +3,14 @@
...
+fi
+
+helm init --wait --service-account tiller --tiller-image gcr.io/kubernetes-helm/tiller:v${HELM_VERSION} --history-max=${HISTORY_MAX} --kube-context ${CLUSTER_NAME}-admin "${params[@]}"
```
Comment (timja) [inline scripts/install-tiller.sh]:
```suggestion
helm init --wait --service-account tiller --tiller-image gcr.io/kubernetes-helm/tiller:v${HELM_VERSION} --history-max=${HISTORY_MAX} --kube-context ${CLUSTER_NAME}-admin "${helm_tls_params[@]}"
```

**Pull request merged**